src/mame/nintendo/n64_v.cpp

```diff
@@ -1358,368 +1360,370 @@ void n64_rdp::disassemble(uint64_t *cmd_buf, char* buffer)
-			sprintf(dxmdy,  "%4.4f", (float)int32_t(cmd_buf[3])       / 65536.0f);
-
-			sprintf(buffer, "Tri_NoShade            %d, XL: %s, XM: %s, XH: %s, YL: %s, YM: %s, YH: %s\n", lft, xl,xm,xh,yl,ym,yh);
+			snprintf(yl,     std::size(yl),    "%4.4f", (float)((cmd_buf[0] >> 32) & 0x1fff) / 4.0f);
+			snprintf(ym,     std::size(ym),    "%4.4f", (float)((cmd_buf[0] >> 16) & 0x1fff) / 4.0f);
+			snprintf(yh,     std::size(yh),    "%4.4f", (float)((cmd_buf[0] >>  0) & 0x1fff) / 4.0f);
+			snprintf(xl,     std::size(xl),    "%4.4f", (float)int32_t(cmd_buf[1] >> 32) / 65536.0f);
+			snprintf(dxldy,  std::size(dxldy), "%4.4f", (float)int32_t(cmd_buf[1])       / 65536.0f);
+			snprintf(xh,     std::size(xh),    "%4.4f", (float)int32_t(cmd_buf[2] >> 32) / 65536.0f);
+			snprintf(dxhdy,  std::size(dxhdy), "%4.4f", (float)int32_t(cmd_buf[2])       / 65536.0f);
+			snprintf(xm,     std::size(xm),    "%4.4f", (float)int32_t(cmd_buf[3] >> 32) / 65536.0f);
+			snprintf(dxmdy,  std::size(dxmdy), "%4.4f", (float)int32_t(cmd_buf[3])       / 65536.0f);
+
+			util::stream_format(buffer, "Tri_NoShade            %d, XL: %s, XM: %s, XH: %s, YL: %s, YM: %s, YH: %s\n", lft, xl,xm,xh,yl,ym,yh);
 			break;
 		}
 		case 0x09:      // Tri_NoShadeZ
 		{
 			const int32_t lft = (cmd_buf[0] >> 55) & 0x1;
 
-			sprintf(yl,     "%4.4f", (float)((cmd_buf[0] >> 32) & 0x1fff) / 4.0f);
-			sprintf(ym,     "%4.4f", (float)((cmd_buf[0] >> 16) & 0x1fff) / 4.0f);
-			sprintf(yh,     "%4.4f", (float)((cmd_buf[0] >>  0) & 0x1fff) / 4.0f);
-			sprintf(xl,     "%4.4f", (float)int32_t(cmd_buf[1] >> 32) / 65536.0f);
-			sprintf(dxldy,  "%4.4f", (float)int32_t(cmd_buf[1])       / 65536.0f);
-			sprintf(xh,     "%4.4f", (float)int32_t(cmd_buf[2] >> 32) / 65536.0f);
-			sprintf(dxhdy,  "%4.4f", (float)int32_t(cmd_buf[2])       / 65536.0f);
-			sprintf(xm,     "%4.4f", (float)int32_t(cmd_buf[3] >> 32) / 65536.0f);
-			sprintf(dxmdy,  "%4.4f", (float)int32_t(cmd_buf[3])       / 65536.0f);
-
-			sprintf(buffer, "Tri_NoShadeZ            %d, XL: %s, XM: %s, XH: %s, YL: %s, YM: %s, YH: %s\n", lft, xl,xm,xh,yl,ym,yh);
+			snprintf(yl,     std::size(yl),    "%4.4f", (float)((cmd_buf[0] >> 32) & 0x1fff) / 4.0f);
+			snprintf(ym,     std::size(ym),    "%4.4f", (float)((cmd_buf[0] >> 16) & 0x1fff) / 4.0f);
+			snprintf(yh,     std::size(yh),    "%4.4f", (float)((cmd_buf[0] >>  0) & 0x1fff) / 4.0f);
+			snprintf(xl,     std::size(xl),    "%4.4f", (float)int32_t(cmd_buf[1] >> 32) / 65536.0f);
+			snprintf(dxldy,  std::size(dxldy), "%4.4f", (float)int32_t(cmd_buf[1])       / 65536.0f);
+			snprintf(xh,     std::size(xh),    "%4.4f", (float)int32_t(cmd_buf[2] >> 32) / 65536.0f);
+			snprintf(dxhdy,  std::size(dxhdy), "%4.4f", (float)int32_t(cmd_buf[2])       / 65536.0f);
+			snprintf(xm,     std::size(xm),    "%4.4f", (float)int32_t(cmd_buf[3] >> 32) / 65536.0f);
+			snprintf(dxmdy,  std::size(dxmdy), "%4.4f", (float)int32_t(cmd_buf[3])       / 65536.0f);
```

Comment:
In general, I'm happy to take this file further and remove some of the code duplication and all the snprintfs in the process.
What od you think, as part of this PR or a separate one?

Reply:
Yeah, it can be cleaned up further, but it can wait.